### PR TITLE
chore: Fix linter findings for `revive:exported` in `plugins/inputs/l*`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Summary

Address findings for [revive:exported ](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#exported ) in `plugins/inputs/l*`.

As part of this effort for files from `plugins/inputs/l*`, the following actions were taken:
- Type names (`const`, `var`, `struct`, `func`, etc) were changed to unexported, wherever they didn't need to be exported.
- All remaining exported types were given comments in the appropriate form – this does not apply to exported methods that implement "known" plugin interfaces (`Gather|Init|Start|Stop|SampleConfig|Parse|Add|Apply|Serialize|SerializeBatch|SetParser|SetParserFunc|GetState|SetState`).
- The order of methods was organized (exported methods first, then unexported, with `init` at the very end).

It is only part of the bigger work (for issue: https://github.com/influxdata/telegraf/issues/15813).
After all findings of this type in whole project are handled, we can enable `revive:exported` rule in `golangci-lint`.

In addition:

- `plugins/inputs/linux_sysctl_fs/linux_sysctl_fs.go` contained a method that extracted the path to the `proc` directory from `HOST_PROC` (or returned the default: `/proc`). This method was used by another plugin, `plugins/inputs/processes/processes_notwindows.go`. I moved this method to `internal/env.go` and adjusted the code of all plugins which did similar things to use it. I did the same for `HOST_SYS`.
- While changing methods from `exported` to `unexported` in the interface implemented by `mock`, I regenerated all mocks using the latest `mockery` (for input plugins from `a*` to `l*`).

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/inputs/lanz/lanz.go [hunk 1]
```diff
@@ -130,3 +120,13 @@ func msgToAccumulator(acc telegraf.Accumulator, msg *pb.LanzRecord, deviceURL *u
 		acc.AddFields("lanz_global_buffer_usage_record", vals, tags)
 	}
 }
+
+func newLanz() *Lanz {
+	return &Lanz{}
+}
```
Comment (srebhan) [inline plugins/inputs/lanz/lanz.go:126]: Do you think we can get rid of this function in the present PR?

File: plugins/inputs/logstash/logstash.go [hunk 2]
```diff
@@ -122,6 +122,69 @@ func (logstash *Logstash) Init() error {
 	return nil
 }
 
+func (logstash *Logstash) Start(_ telegraf.Accumulator) error {
+	return nil
+}
```
Comment (srebhan) [inline plugins/inputs/logstash/logstash.go:127]:
```suggestion
func (*Logstash) Start(telegraf.Accumulator) error {
	return nil
}
```

**Review by srebhan — APPROVED** [review]
Thanks @zak-pawel! Just one comment about getting rid of a useless function. Approving anyway in case you want to do it in a separate PR...

File: plugins/inputs/lanz/lanz.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/lanz/lanz.go]: Removed.

File: plugins/inputs/logstash/logstash.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/inputs/logstash/logstash.go]: Changed.

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**